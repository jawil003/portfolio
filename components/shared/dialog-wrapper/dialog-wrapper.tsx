import { css } from "@emotion/react";
import { UniversalPortal } from "@jesstelford/react-portal-universal";
import React from "react";

export interface Props {
  show: boolean;
}

/**
 * An DialogWrapper React Component.
 * @author Jannik Will
 * @version 0.1
 */
export const DialogWrapper: React.FC<Props> = ({
  show,
  children,
}) => {
  return (
    <UniversalPortal selector="#modal">
      <div>
        <div
          css={css`
            & {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              min-width: 100vw;
              max-width: 100vw;
              width: 100vw;
              height: 100vh;
              min-height: 100vh;
              max-height: 100vh;
              margin: 0px;
              padding: 0px;
              box-sizing: border-box;
              justify-content: center;
              align-items: center;
              display: ${show
                ? "flex"
                : "none"};
              flex-direction: column;
            }
            &:after {
              content: "";
              position: absolute;
              top: 0;
              left: 0;
              min-width: 100vw;
              max-width: 100vw;
              width: 100vw;
              height: 100vh;
              min-height: 100vh;
              max-height: 100vh;
              background-color: #000000;
              opacity: 0.2;

              z-index: 1;
            }
          `}
        >
          <div
            css={css`
              & {
                box-sizing: border-box;
                background-color: var(
                  --greenblue
                );
                min-width: 400px;
                min-height: 200px;
                border-radius: 10px;
                z-index: 999;
                flex-direction: column;
                padding: 20px;
                margin: -10px 0px;
              }

              & > * {
                margin: 10px 0px;
                color: var(--white);
              }
            `}
          >
            {children}
          </div>
        </div>
      </div>
    </UniversalPortal>
  );
};
