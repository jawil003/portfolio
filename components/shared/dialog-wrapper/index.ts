export { DialogWrapper } from "./dialog-wrapper";
export type { Props as DialogWrapperProps } from "./dialog-wrapper";
