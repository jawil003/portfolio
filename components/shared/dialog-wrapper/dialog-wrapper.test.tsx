import {
  render,
  RenderResult,
} from "@testing-library/react";
import { DialogWrapper } from "./dialog-wrapper";

describe("DialogWrapper", () => {
  let testDialogWrapper: RenderResult<
    typeof import("@testing-library/dom/types/queries"),
    HTMLElement
  >;
  beforeEach(() => {
    document.body.innerHTML =
      '<div id="modal" />';

    testDialogWrapper = render(
      <DialogWrapper show />,
    );
  });
  test("Create DialogWrapper", () => {
    expect(
      testDialogWrapper.baseElement,
    ).toMatchSnapshot();
  });
});
