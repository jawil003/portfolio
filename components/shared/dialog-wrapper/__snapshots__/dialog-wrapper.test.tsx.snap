// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogWrapper Create DialogWrapper 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  min-width: 100vw;
  max-width: 100vw;
  width: 100vw;
  height: 100vh;
  min-height: 100vh;
  max-height: 100vh;
  margin: 0px;
  padding: 0px;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0:after {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  min-width: 100vw;
  max-width: 100vw;
  width: 100vw;
  height: 100vh;
  min-height: 100vh;
  max-height: 100vh;
  background-color: #000000;
  opacity: 0.75;
  z-index: 1;
}

.emotion-1 {
  box-sizing: border-box;
  background-color: var(
                    --greenblue
                  );
  min-width: 400px;
  min-height: 200px;
  border-radius: 10px;
  z-index: 999;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
  margin: -10px 0px;
}

.emotion-1>* {
  margin: 10px 0px;
  color: var(--white);
}

<body>
  <div
    id="modal"
  >
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1"
        />
      </div>
    </div>
  </div>
  <div />
</body>
`;
